dfsclient/src/test/java/org/hpccsystems/dfs/client/DFSHPCCFile.java

```diff
@@ -192,6 +192,43 @@ public final void testGetProjectedRecordDefinition() throws HpccFileException
         Assert.assertEquals(3, projectedRecordDefinition.getNumDefs());
     }
 
+    private boolean testFieldDefsNotEquals(FieldDef fieldDef1, FieldDef fieldDef2)
+    {
+        for (int i = 0; i < fieldDef1.getNumDefs(); i++)
+        {
+            if (fieldDef1.getDef(i) == fieldDef2.getDef(i))
+            {
+                return false;
+            }
+            
+            if (!testFieldDefsNotEquals(fieldDef1.getDef(i), fieldDef2.getDef(i)))
```

Comment:
what is this second test doing?

Reply:
We need to recursively check child field defs for things like embedded records, child datasets, etc.

Reply:
Figured as much, but didn't see where the child structure was passed in, "getDef(i)" wasn't explicit enough for me